src/core/initializeComponentsFactory.js

```diff
@@ -80,12 +80,12 @@ export default (
 
   const lifecycle = createLifecycle(componentRegistry);
   const network = createNetwork(config, logger, lifecycle);
-  lifecycle.onComponentsRegistered({
-    componentRegistry,
-    lifecycle,
-    network,
-    optIn
-  });
```

Comment:
This is why errors in `onComponentsRegistered` weren't being suppressed. Nothing was handling errors from this promise.